### PR TITLE
Add dependency `transformers` to extra-ml-requirements, `virtualenv` to test-requirements

### PR DESCRIPTION
**Pull request by WeichenXu123**
Signed-off-by: Weichen Xu <weichen.xu@databricks.com>

## Related Issues/PRs

#xxx

## What changes are proposed in this pull request?

Add dependency `transformers` to extra-ml-requirements, `virtualenv` to test-requirements

## How is this patch tested?

- [ ] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

Rerun unit tests.

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Click the `Details` link on the `Preview docs` check.
2. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [x] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: requirements/test-requirements.txt [hunk 1]
```diff
@@ -19,3 +19,4 @@ plotly
 kaleido
 # The latest flask version requires werkzeug>=2.2.0
 werkzeug>=2.2.0
+virtualenv
```
Comment (harupy) [inline requirements/test-requirements.txt:22]: `virtualenv` is included in the extra dependencies.

https://github.com/mlflow/mlflow/blob/2305eb5c2397f5739a10bda559790be0b5a05ad9/setup.py#L147

File: requirements/extra-ml-requirements.txt [hunk 2]
```diff
@@ -50,3 +50,5 @@ prophet
 pmdarima
 # Required by mlflow.diviner
 diviner
+# Required by mlflow.tensorflow
```
Comment (harupy) [inline requirements/extra-ml-requirements.txt:53]: This sound like `mlflow.tensorflow` uses `transformers`, but it doesn't.

Comment (harupy) [reply requirements/extra-ml-requirements.txt:53]: Only a few tests require transformers. Do we really need to include it in `requirements/extra-ml-requirements.txt`?

Comment (WeichenXu123) [reply requirements/extra-ml-requirements.txt:53]: ok, what about moving transformers into test requirements ?
A test requires it.

File: requirements/test-requirements.txt [hunk 3]
```diff
@@ -19,3 +19,5 @@ plotly
 kaleido
 # The latest flask version requires werkzeug>=2.2.0
 werkzeug>=2.2.0
+# Required by mlflow.tensorflow tests.
```
Comment (harupy) [inline requirements/test-requirements.txt:22]: Are you sure about this?

image

Comment (harupy) [reply requirements/test-requirements.txt:22]: image

Comment (harupy) [reply requirements/test-requirements.txt:22]: `transformers` is used in `tests/pytorch/test_pytorch_model_export.py` and `tests/keras/test_keras_model_export.py`.

Comment (harupy) [reply requirements/test-requirements.txt:22]: https://github.com/mlflow/mlflow/blob/2305eb5c2397f5739a10bda559790be0b5a05ad9/mlflow/ml-package-versions.yml#L34-L38

Comment (harupy) [reply requirements/test-requirements.txt:22]: https://github.com/mlflow/mlflow/blob/2305eb5c2397f5739a10bda559790be0b5a05ad9/mlflow/ml-package-versions.yml#L97-L109

Comment (harupy) [reply requirements/test-requirements.txt]: @WeichenXu123 `ml-package-versions.yml` already contains `transformers`.

Comment (WeichenXu123) [reply requirements/test-requirements.txt]: shall we also added in test_requirements so that setting up development environment can include it ?

**Review by harupy — APPROVED** [review]

**Pull request merged**